src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -860,13 +860,18 @@ internal void ValidateCopyToArguments(Stream destination, int bufferSize)
                 throw new ArgumentNullException("destination");
             if (bufferSize <= 0)
                 throw new ArgumentOutOfRangeException("bufferSize", Environment.GetResourceString("ArgumentOutOfRange_NeedPosNum"));
-            if (!CanRead && !CanWrite)
+
+            // Cache some virtual method calls for better perf
+            bool canRead = CanRead;
+            bool destCanWrite = destination.CanWrite;
```

Comment:
I noticed that we were calling `CanRead` / `destination.CanWrite` twice during argument validation, so I cached them into locals here.